src/docs/sphinx/dev_guide/gpu_porting.rst

```diff
@@ -0,0 +1,297 @@
+.. ## Copyright (c) 2017-2022, Lawrence Livermore National Security,con LLC and
+.. ## other Axom Project Developers. See the top-level LICENSE file for details.
+.. ##
+.. ## SPDX-License-Identifier: (BSD-3-Clause)
+
+.. _gpu-porting-label:
+
+*******************************
+GPU Porting in Axom
+*******************************
+
+Axom uses RAJA and Umpire as the main workhorses for GPU porting,
+with a set of convenience macros, Axom name-spaced wrappers around commonly
+used functions, and preset execution spaces for host/device execution.
+
```

Comment:
It might be good to mention something about our kernel and memory space support when axom is configured without RAJA and Umpire respectively. (It doesn't have to be here though).

Specifically, `axom::forall` works and so do our memory functions, but they only work on the default space (i.e. `host`/CPU) and with the std allocator.